File: src/main/java/net/legitimoose/bot/config/LegitimooseBotConfigModel.java
package net.legitimoose.bot.config;

import io.wispforest.owo.config.annotation.Config;

@Config(name = "legitimoosebot-config", wrapperName = "LegitimooseBotConfig")
public class LegitimooseBotConfigModel {
    public String webhookUrl = "";
    public String secretPrefix = "::";
    public String mongoUri = "";
    public String discordToken = "";
    public String channelId = "";
    public int waitMinutesBetweenScrapes = 10;
    public boolean verboseLogging = false;
}

File: src/main/kotlin/net/legitimoose/bot/config/LegitimooseBotConfigModel.kt
package net.legitimoose.bot.config

import io.wispforest.owo.config.annotation.Config

@Config(name = "legitimoosebot-config", wrapperName = "LegitimooseBotConfig")
class LegitimooseBotConfigModel {
    @JvmField var webhookUrl = ""

    @JvmField var errorWebhookUrl = ""

    @JvmField var secretPrefix = "::"

    @JvmField var mongoUri = ""

    @JvmField var discordToken = ""

    @JvmField var channelId = ""

    @JvmField var waitMinutesBetweenScrapes: Long = 10
}
